x-pack/plugins/observability_solution/observability_ai_assistant/server/service/util/catch_function_limit_exceeded_error.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { catchError, filter, of, OperatorFunction, shareReplay, throwError } from 'rxjs';
+import {
+  ChatCompletionChunkEvent,
+  MessageAddEvent,
+  MessageRole,
+  StreamingChatResponseEventType,
+} from '../../../common';
+import { isFunctionNotFoundError } from '../../../common/conversation_complete';
+import { emitWithConcatenatedMessage } from '../../../common/utils/emit_with_concatenated_message';
+
+export function catchFunctionLimitExceededError(): OperatorFunction<
```

Comment:
This removes the function call from the message if the function limit has been exceeded and the LLM tries to call a function anyway. It's a last resort.